core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SPARQLProtocolSession.java

```diff
@@ -1041,6 +1044,10 @@ protected HttpResponse execute(HttpUriRequest method) throws IOException, RDF4JE
 						throw new MalformedQueryException(errInfo.getErrorMessage());
 					} else if (errInfo.getErrorType() == ErrorType.UNSUPPORTED_QUERY_LANGUAGE) {
 						throw new UnsupportedQueryLanguageException(errInfo.getErrorMessage());
+					} else if (exceptionIs(response, "ShaclSailValidationException")) {
```

Comment:
I can try to move it there and see if it works.

Reply:
Took a look now. This execute(....) method is what is used by RDF4JProtocolSession for its requests. RDF4JProtocolSession extends SPARQLProtocolSession. We should probably extract an even lower class that only implements execute(....) so that we can be sure that no one above can bypass it. This isn't in scope for this PR though.

Reply:
Fair enough, let's leave for now. It's not in the way.